python/ray/data/dataset.py

```diff
@@ -3930,7 +3930,32 @@ def __iter__(self):
         return pipe
 
     def fully_executed(self) -> "Dataset[T]":
-        """Force full evaluation of the blocks of this dataset.
+        """Deprecated: use `ds.cache()` instead"""
```

Comment:
nit: shall we be more explicit with changing `ds.cache()` to `Dataset.cache()` or `cache()`? Technically there's no such a thing of `ds`.